django/core/management/commands/migrate.py

```diff
@@ -237,7 +237,7 @@ def handle(self, *args, **options):
                     self.stdout.write("  No migrations to prune.")
 
         plan = executor.migration_plan(targets)
-        exit_dry = plan and options["check_unapplied"]
+        exit_dry = options["check_unapplied"]
```

Comment:
I don't thinks it's a proper patch, `--check` should return when all migrations are applied, not raise a `SystemExit`.

Reply:
As far as I'm aware more changes are needed, e.g.
```diff
diff --git a/django/core/management/commands/migrate.py b/django/core/management/commands/migrate.py
index d98072c66e..f898f7f342 100644
--- a/django/core/management/commands/migrate.py
+++ b/django/core/management/commands/migrate.py
@@ -237,22 +237,24 @@ class Command(BaseCommand):
                     self.stdout.write("  No migrations to prune.")
 
         plan = executor.migration_plan(targets)
-        exit_dry = plan and options["check_unapplied"]
-
+        check_unapplied = options["check_unapplied"]
+        if check_unapplied and not plan:
+            return
         if options["plan"]:
             self.stdout.write("Planned operations:", self.style.MIGRATE_LABEL)
             if not plan:
                 self.stdout.write("  No planned migration operations.")
-            for migration, backwards in plan:
-                self.stdout.write(str(migration), self.style.MIGRATE_HEADING)
-                for operation in migration.operations:
-                    message, is_error = self.describe_operation(operation, backwards)
-                    style = self.style.WARNING if is_error else None
-                    self.stdout.write("    " + message, style)
-            if exit_dry:
-                sys.exit(1)
+            else:
+                for migration, backwards in plan:
+                    self.stdout.write(str(migration), self.style.MIGRATE_HEADING)
+                    for operation in migration.operations:
+                        message, is_error = self.describe_operation(operation, backwards)
+                        style = self.style.WARNING if is_error else None
+                        self.stdout.write("    " + message, style)
+                if check_unapplied:
+                    sys.exit(1)
             return
-        if exit_dry:
+        if check_unapplied and plan:
             sys.exit(1)
         if options["prune"]:
             return
```

Reply:
(Sorry for the hasty suggestion @na86421.)

Reply:
@felixxm 
Thanks for the advice. I will modify the code.

@jacobtylerwalls 
That's okay :) I was the owner of this ticket, so I had to look more carefully. 

Thank you.

Reply:
I changed it by referring to the comments.